backend/core/llm_chain.py

```
@@ -32,93 +34,230 @@
+        self._lock = RLock()
+
+    def is_rate_limited(self, provider: str) -> bool:
+        """Check if a provider is currently rate limited"""
+        with self._lock:
+
+            if provider not in self._rate_limit_status:
+                return False
+
+            # Check if rate limit has expired
+            if provider in self._rate_limit_reset_time:
+                if datetime.now() > self._rate_limit_reset_time[provider]:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)